### ISSUE TITLE
SDG: table of contents "index"

### ISSUE THREAD

**Issue by mlbrgl** [owid/owid-grapher#1733]
<img width="872" alt="Screenshot 2022-10-28 at 09 23 25" src="https://user-images.githubusercontent.com/13406362/198528144-1d5a94fd-54a5-45f4-9f11-117d88b8f0a0.png">

🎨 [figma prototype](https://www.figma.com/proto/9bQW46okOjxeJtismQQdPu/%5BDesktop%5D-Screens-1.0?node-id=2052%3A18791&scaling=min-zoom&page-id=1829%3A17659&starting-point-node-id=1829%3A19309)

- [x] extract headings into structured JSON
- [x] style
- [ ] `won't do` responsive behaviour (blocked by grid refactor)
- [x] open / close behaviour
- [x] supertitle support ("Target 15.1")

**Comment by mlbrgl** [comment]
Reverted attempt to get the ToC from the generated google docs ToC

<img width="567" alt="Screenshot 2022-10-28 at 18 55 46" src="https://user-images.githubusercontent.com/13406362/198691658-16f9f21c-7851-457b-86f2-273081538679.png">

The initial idea was to give more power to the author if they wanted to amend the automatic generation, by simply editing the  automatically generated content. Syncing the heading slug between the headings on the page (which need the slug as an anchor target) and the table of content entries seemed not worth the potential gain.
